packages/blade/src/components/Tabs/_decisions/decisions.md

```diff
@@ -0,0 +1,385 @@
+
+<img src="./tabs-thumbnail.png" width="380" alt="Tabs thumbnail" />
+
+- [Design](#design)
+- [Features](#features)
+- [Anatomy](#anatomy)
+- [`Tabs` API](#tabs-api)
+- [`TabItem` API](#tabitem-api)
+- [`TabPanel` API](#tabpanel-api)
+  - [Examples:](#examples)
+- [Basic](#basic)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Vertical Orientation](#vertical-orientation)
+- [Filled Variant](#filled-variant)
+- [With Tooltip](#with-tooltip)
+- [Alternative APIs](#alternative-apis)
+  - [1. Quite tricky/hack to implement](#1-quite-trickyhack-to-implement)
+  - [2. The JSX doesn't map well with the visual hierarchy / the HTML structure](#2-the-jsx-doesnt-map-well-with-the-visual-hierarchy--the-html-structure)
+  - [3. Extensibility](#3-extensibility)
+  - [4. API Familiarity](#4-api-familiarity)
+- [Motion](#motion)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?node-id=3733%3A8981&mode=dev) to all variants of the Tabs component
+
+## Features
+
+- Controlled and Uncontrolled
+- Filled and Bordered variants
+- Horizontal and Vertical orientation
+- Medium and Large size
+
+## Anatomy
+
+<img src="./tabs-anatomy.png" width="100%" alt="Tabs thumbnail" />
+
+```jsx
+<Tabs>
+  <TabList>
+    <TabItem />
+  </TabList>
+  <TabPanel />
+</Tabs>
+```
+
+## `Tabs` API
+
+```ts
+type TabsProps = {
+  /**
+   * The content of the component, accepts `TabList` and `TabPanel` components.
+   */
+  children: React.ReactNode;
+  /**
+   * The value of the selected tab, If set the component will be controlled.
+   */
+  value?: string;
+  /**
+   * The default value of the selected tab, in case the Tabs component is uncontrolled.
+   */
+  defaultValue?: string;
+  /**
+   * Callback fired when the value changes.
+   */
+  onChange?: (value: string) => void;
+  /**
+   * The orientation of the tabs.
+   */
+  orientation?: 'horizontal' | 'vertical';
+  /**
+   * The size of the tabs.
+   */
+  size?: 'medium' | 'large';
+  /**
+   * The variant of the tabs.
+   */
+  variant?: 'bordered' | 'filled';
+  /**
+   * If `true`, the TabItems will grow to use all the available space.
+   */
+  autoWidth?: boolean;
+};
+```
+
+## `TabItem` API
+
+```ts
+type TabItemProps = {
+  /**
+   * The label of the tab item.
+   */
+  children: string;
+  /**
+   * The value of the tab item.
+   */
+  value: string;
+  /**
+   * Leading element of the tab item.
+   * Can be used to render an icon.
+   */
+  leading?: React.ReactNode;
+  /**
+   * Trailing element of the tab item.
+   * Can be used to render an badge/counter.
+   */
+  trailing?: React.ReactNode;
+  /**
+   * If `true`, the tab item will be disabled.
+   */
+  isDisabled?: boolean;
+};
+```
+
+## `TabPanel` API
+
```

Comment:
Can we employ lazy loading for various tab panels? Maybe an example will help. Also will the tab panels be unmounted and removed from DOM when unselected?

Reply:
🤔 Hmm. 

> Could probably expose a lazy prop
When you say "lazy loading tabs" do you mean to have the Tab's content be a Suspense component? Or do you mean a "lazy" prop which will control the mounted behaviour of the tabs, eg: lazy=true ---> unmounts on selection change, lazy=false ---> doesn't unmount just hides via display: none.

> Also will the tab panels be unmounted and removed from DOM when unselected?
right now we unmount unselected tabs.

Reply:
>right now we unmount unselected tabs.

this could be trickier because in some product cases we might need to retain the state of the children(for eg checkbox or an accordion in the previous selected tab)

Reply:
> this could be trickier because in some product cases we might need to retain the state of the children(for eg checkbox or an accordion in the previous selected tab)

This can be solved, by adding lazy prop, on react-native side, [react-native-tab-view](https://reactnavigation.org/docs/tab-view/) also provides a similar API. Should be simple enough to implement.